lib/util/sdp/trackmatcher/mid.js

```diff
@@ -0,0 +1,47 @@
+'use strict';
+
+var getMediaSections = require('../').getMediaSections;
+
+/**
+ * Construct an {@link MIDTrackMatcher}.
+ * @class
+ * @classdesc An {@link MIDTrackMatcher} matches an RTCTrackEvent with a
+ *   MediaStreamTrack ID based on the MID of the underlying RTCRtpTransceiver.
+ */
+function MIDTrackMatcher() {
+  if (!(this instanceof MIDTrackMatcher)) {
+    return new MIDTrackMatcher();
+  }
+  Object.defineProperties(this, {
+    _midToTrackIds: {
+      value: new Map(),
+      writable: true
+    }
+  });
+}
+
+/**
+ * Match a given MediaStreamTrack with its ID.
+ * @param {RTCTrackEvent} event
+ * @returns {Track.ID}
+ */
+MIDTrackMatcher.prototype.match = function match(event) {
+  return this._midToTrackIds.get(event.transceiver.mid) || null;
+};
+
+/**
+ * Update the {@link MIDTrackMatcher} with a new SDP.
+ * @param {string} sdp
+ */
+MIDTrackMatcher.prototype.update = function update(sdp) {
+  var sections = getMediaSections(sdp, '(audio|video)');
+  this._midToTrackIds = sections.reduce(function(midToTrackIds, section) {
+    var midMatches = section.match(/^a=mid:(.+)$/m) || [];
+    var trackIdMatches = section.match(/^a=msid:.+ (.+)$/m) || [];
+    var mid = midMatches[1];
+    var trackId = trackIdMatches[1];
+    return mid && trackId ? midToTrackIds.set(mid, trackId) : midToTrackIds;
+  }, this._midToTrackIds);
```

Comment:
Nitpick, but we are mapping to a singular `trackId`, right? So should the collection and private property here be `_midToTrackId`? IDK if we are very strict on this pattern throughout the codebase. Just a thought...

Reply:
@markandrus I think we follow this nomenclature: `midsToTrackIds`. Since we're describing a collection, I think the name should be plural. WDYT?